### ISSUE TITLE
Fixes compiling Python API on Mac OS X 10.12 El Capitan

### ISSUE THREAD

**Issue by avitale** [davisking/dlib#255]
I had 2 errors compiling Python API on Mac OS X 10.12 El Capitan, here is how I fixed them in case it could help others.

Error 1:

```
fatal error: 'initializer_list' file not found
    #include <initializer_list>
             ^
    1 error generated.
    make[2]: *** [dlib_build/CMakeFiles/dlib.dir/data_io/image_dataset_metadata.cpp.o] Error 1 
```

Fix 1:

Add  environment variable `MACOSX_DEPLOYMENT_TARGET=10.12`

`MACOSX_DEPLOYMENT_TARGET=10.12 python setup.py install`

Error 2:

```
fatal error: 'gif_lib.h' file not found
#include <gif_lib.h>
```

Fix 2:

`brew install giflib`

**Comment by davisking** [comment]
When I try to build the Python API on Mac OS X 10.11.5 I don't get these errors.   I suspect very much that at least the libgif one is due to another broken libgif on your system.  Can you look in cmake and see what's happening?  If libgif isn't installed then it should simply be not used and no error should be generated.  To get the error you got cmake must have found a libgif install, but then that install was subsequently broken, leading to the gif_lib.h file not found error.   

Therefore, the best way to fix this is to figure out how to add something to the cmake script that detects whatever causes cmake to think there is a working libgif installed when there in fact isn't.  


The error about initializer_list not found is also bizarre.  That is a standard C++ header file.  On the OS X machines I have it is present and does not require any external variables to be set like MACOSX_DEPLOYMENT_TARGET.  Are you sure there is not something funny going on in your system that causes this?

**Comment by avitale** [comment]
For libgif, some software like Mono installed it a dependency. I can't reproduce the error after I installed it with homebrew.

The error about initializer_list is probably due to the fact that I updated El capitan from an older OS X. Before setting the flag, the compiler was automatically looking for an old MacOSX.platform SDK that wasn't anymore available, after setting the flag it looks for the available MacOSX10.12.sdk
